domains/pallets/messenger/src/lib.rs

```diff
@@ -46,6 +46,10 @@ use sp_messenger::messages::{
 use sp_runtime::traits::{Extrinsic, Hash};
 use sp_runtime::DispatchError;
 
+/// Maximum bound of number of channels we close if a chain is removed from the allowlist.
+/// This is not a Hard limit but rather used to find the weight for extrinsic.
+pub const MAX_CHANNELS_TO_CLOSE: u32 = 100;
```

Comment:
This may just defeat the purpose of extrinsic weight which is used to ensure the extrinsic execution will finish within the max expected time. Since we don't really have a limit on the number of channels, if there are 1000 channels and an extrinsic comes to remove the domain from allowlist, it will take 10x more time to execute than expected and cause slow block execution.

Closing the channel in async is possible IMHO, we just need to add a check when verifying XDM to ensure the src chain id is still in the allowlist, and the channel owner is incentivized to close the channel by themself to get their `ChannelReserveFee` back.

Reply:
> This may just defeat the purpose of extrinsic weight which is used to ensure the extrinsic execution will finish within the max expected time. Since we don't really have a limit on the number of channels, if there are 1000 channels and an extrinsic comes to remove the domain from allowlist, it will take 10x more time to execute than expected and cause slow block execution.

That is fair and thanks for bringing this up

> Closing the channel in async is possible IMHO, we just need to add a check when verifying XDM to ensure the src chain id is still in the allowlist, and the channel owner is incentivized to close the channel by themself to get their ChannelReserveFee back.

This IMHO is not a correct approach as well. We need to ensure the flow between request and response continues albeit Err or OK. I have described more above.